### PR TITLE
Don't show the "Leave section" button to students in OAuth sections

### PR DESCRIPTION
**Pull request by joshlory**
image

+cc @poorvasingal

### CODE REVIEW

**Comment by joshlory** [comment]
This component still shows the `G-` section code.  My understanding was that this component will be merged with the proper `SectionTable` component?  Cleaner to implement this via `providerManaged` if these components are not going to be merged in the future.

**Comment by Bjvanminnen** [comment]
My understanding (which may be imperfect) is that SectionsTable will go away after the section-flow-2017 experiment becomes enabled. Presumably that implies we'll want to make this same sort of change there?

File: apps/src/templates/studioHomepages/SectionsTable.jsx [hunk 1]
```diff
@@ -203,12 +203,14 @@ const SectionsTable = React.createClass({
               </td>
               {!isTeacher && canLeave && (
                 <td style={{...styles.col, ...styles.leaveCol}}>
-                  <Button
-                    style={{marginLeft: 5}}
-                    text={i18n.leaveSection()}
-                    onClick={this.onLeave.bind(this, section.code, section.name)}
-                    color={Button.ButtonColor.gray}
-                  />
+                  {!/^(C|G)-/.test(section.code) &&
```
Comment (Bjvanminnen) [inline apps/src/templates/studioHomepages/SectionsTable.jsx:206]: I might make a helper method called something like `isOauthSection(section)` that can include a comment about what this regex is doing/why we can only leave oauth sections.

**Comment by Bjvanminnen** [comment]
Apparently SectionsTable will not go away completely, as it will still be used by the `StudentSections` component.

**Comment by joshlory** [comment]
@poorvasingal does this mean we need to hide the `G-` and `C-` codes for the `SectionsTable` component?  I was asked not to do that work because it was going to be duplicating logic in the `SectionTable` component, which was to replace this component on the Studio home page.

**Pull request merged**